### ISSUE TITLE
mapping generator error

### ISSUE THREAD

**Issue by taterbase** [mongoosastic/mongoosastic#3]
https://github.com/jamescarr/mongoosastic/issues/65#issue-13299454
>Hello there. Thanks for this fantastic project, I noticed a small bug as I updated my schema
mongoosastic fail to generate my schema mapping when my schema has complex attributes like the following:

>```javascript
new Schema({
    recents: {
      list: [{type: ObjectId, ref: 'User'}],
      size: {type: Number}
    }
  });
```
>I don't even need to index this attributes but 
createMapping generate the following error 
````
        else if ( paths[field].caster && paths[field].caster.instance ) {
                              ^
TypeError: Cannot read property 'caster' of undefined
  at getCleanTree (/Users/pg/Jogabo/app/node_modules/mongoosastic/lib/mapping-generator.js:122:31)
  at getCleanTree (/Users/pg/Jogabo/app/node_modules/mongoosastic/lib/mapping-generator.js:148:28)
  at Generator.generateMapping (/Users/pg/Jogabo/app/node_modules/mongoosastic/lib/mapping-generator.js:5:19)
  at createMappingIfNotPresent (/Users/pg/Jogabo/app/node_modules/mongoosastic/lib/mongoosastic.js:142:13)
  at Function.module.exports.schema.statics.createMapping (/Users/pg/Jogabo/app/node_modules/mongoosastic/lib/mongoosastic.js:39:5)
```

**Comment by jaydub2099** [comment]
I've run into the same issue. For what it's worth in my example, one of the fields with a subdocument is a single value field (topic) whereas the other is a multi value field (related_topics). I have no problems with the multi value field (which has a multi value subdocument). However the multi value subdocument within the single value field triggers the error. If I change the multi value subdocument for the single value field (topic) to a single value subdocument, I no longer get the error. In this particular case I can get by with the single value field only having a single parent but there will be other cases where a single value field could have multiple parents. Hopefully that helps...

```
  topic: {
    tid: Number,
    term: String,
    parents: [{
      tid: Number,
      term: String,
      _id: false
    }],
    _id: false
  }, // Term ref, subdocument is required

  related_topics: [{
    tid: Number,
    term: String,
    parents: [{
      tid: Number,
      term: String,
      _id: false
    }],
    _id: false
  }], // Term refs
```

**Comment by guumaster** [comment]
This bug is fixed after #47. Please update your dependency.
